salt/engines/slack.py

```diff
@@ -135,6 +146,25 @@ def fire(tag, msg):
                                 # Trim the ! from the front
                                 # cmdline = _text[1:].split(' ', 1)
                                 cmdline = salt.utils.shlex_split(_text[len(trigger):])
+
+                                # Remove inital salt item from list
+                                if cmdline[0] == 'salt':
```

Comment:
Not sure about this one.  The first bit of the command is the trigger so if !salt is the trigger then is should get removed during the for loop on cmdline.  Seems confusing to have ! be the trigger but also include salt.